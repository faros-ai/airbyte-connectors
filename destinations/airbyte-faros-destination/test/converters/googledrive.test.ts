import {
  destinationWriteTest,
  initMockttp,
  tempConfig,
} from 'faros-airbyte-testing-tools';
import {getLocal} from 'mockttp';

describe('googledrive', () => {
  const mockttp = getLocal({debug: false, recordTraffic: false});
  let configPath: string;

  beforeEach(async () => {
    await initMockttp(mockttp);
    configPath = await tempConfig({
      api_url: mockttp.url,
      log_records: true,
    });
  });

  afterEach(async () => {
    await mockttp.stop();
  });

  test('process records from all streams', async () => {
    await destinationWriteTest({
      configPath,
      catalogPath: 'test/resources/googledrive/catalog.json',
      inputRecordsPath: 'googledrive/all-streams.log',
      checkRecordsData: (records) => expect(records).toMatchSnapshot(),
    });
  });
});
