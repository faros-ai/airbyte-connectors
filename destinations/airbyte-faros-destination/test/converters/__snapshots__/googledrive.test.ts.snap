// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`googledrive process records from all streams 1`] = `
Array [
  "Processed 6 records",
  "Processed records by stream: {\\\\\\"mytestsource__googledrive__activity\\\\\\":2,\\\\\\"mytestsource__googledrive__workspace\\\\\\":1,\\\\\\"mytestsource__googledrive__workspace_users\\\\\\":3}\\"},\\"type\\":\\"LOG\\"}",
  "Would write 13 records",
  "Would write records by model: {\\\\\\"dms_Activity\\\\\\":2,\\\\\\"dms_Document\\\\\\":2,\\\\\\"dms_Location\\\\\\":2,\\\\\\"dms_Membership\\\\\\":3,\\\\\\"dms_Organization\\\\\\":1,\\\\\\"dms_User\\\\\\":3}\\"},\\"type\\":\\"LOG\\"}",
  "Skipped 0 records",
  "Errored 0 records",
]
`;

exports[`googledrive process records from all streams 2`] = `
Array [
  Object {
    "dms_Organization": Object {
      "createdAt": "2020-01-01T00:00:00.000Z",
      "domain": "example.com",
      "name": "Example Corp",
      "source": "GoogleDrive",
      "uid": "C01abc234",
    },
  },
  Object {
    "dms_User": Object {
      "createdAt": "2023-06-01T09:00:00.000Z",
      "email": "john.doe@example.com",
      "name": "John Doe",
      "source": "GoogleDrive",
      "type": Object {
        "category": "User",
        "detail": null,
      },
      "uid": "123456789012345678901",
    },
  },
  Object {
    "dms_Membership": Object {
      "organization": Object {
        "source": "GoogleDrive",
        "uid": "C01abc234",
      },
      "source": "GoogleDrive",
      "user": Object {
        "source": "GoogleDrive",
        "uid": "123456789012345678901",
      },
    },
  },
  Object {
    "dms_User": Object {
      "createdAt": "2023-03-15T12:00:00.000Z",
      "email": "jane.smith@example.com",
      "name": "Jane Smith",
      "source": "GoogleDrive",
      "type": Object {
        "category": "User",
        "detail": null,
      },
      "uid": "987654321098765432109",
    },
  },
  Object {
    "dms_Membership": Object {
      "organization": Object {
        "source": "GoogleDrive",
        "uid": "C01abc234",
      },
      "source": "GoogleDrive",
      "user": Object {
        "source": "GoogleDrive",
        "uid": "987654321098765432109",
      },
    },
  },
  Object {
    "dms_User": Object {
      "createdAt": "2023-01-01T09:00:00.000Z",
      "email": "suspended.user@example.com",
      "name": "Suspended User",
      "source": "GoogleDrive",
      "type": Object {
        "category": "User",
        "detail": null,
      },
      "uid": "555666777888999000111",
    },
  },
  Object {
    "dms_Membership": Object {
      "organization": Object {
        "source": "GoogleDrive",
        "uid": "C01abc234",
      },
      "source": "GoogleDrive",
      "user": Object {
        "source": "GoogleDrive",
        "uid": "555666777888999000111",
      },
    },
  },
  Object {
    "dms_Activity": Object {
      "document": Object {
        "source": "GoogleDrive",
        "uid": "1BxiMVs0XRA5nFMdKvBdBZjgmUUqptlbs74OgvE2upms",
      },
      "source": "GoogleDrive",
      "timestamp": "2024-01-01T12:00:00.000Z",
      "type": Object {
        "category": "Edit",
        "detail": "edit",
      },
      "uid": "d60b6236fc2a634716d06392c585ae485f615066b74ce3d439184ffa44e7f895",
      "user": Object {
        "source": "GoogleDrive",
        "uid": "123456789012345678901",
      },
    },
  },
  Object {
    "dms_Activity": Object {
      "document": Object {
        "source": "GoogleDrive",
        "uid": "2CyiMVs0XRA5nFMdKvBdBZjgmUUqptlbs74OgvE2upms",
      },
      "source": "GoogleDrive",
      "timestamp": "2024-01-01T11:30:00.000Z",
      "type": Object {
        "category": "Create",
        "detail": "create",
      },
      "uid": "05d9ff49638506b3ad92892955c87fb89de061855da934a200bddbd53c5ec6ec",
      "user": Object {
        "source": "GoogleDrive",
        "uid": "567890123456789012345",
      },
    },
  },
  Object {
    "dms_Document": Object {
      "location": Object {
        "source": "GoogleDrive",
        "uid": "people/098765432109876543210",
      },
      "source": "GoogleDrive",
      "title": "Example Document",
      "type": Object {
        "category": "Document",
        "detail": "application/vnd.google-apps.document",
      },
      "uid": "1BxiMVs0XRA5nFMdKvBdBZjgmUUqptlbs74OgvE2upms",
    },
  },
  Object {
    "dms_Document": Object {
      "createdAt": "2024-01-01T11:30:00.000Z",
      "location": Object {
        "source": "GoogleDrive",
        "uid": "people/567890123456789012345",
      },
      "source": "GoogleDrive",
      "title": "Test Spreadsheet",
      "type": Object {
        "category": "Spreadsheet",
        "detail": "application/vnd.google-apps.spreadsheet",
      },
      "uid": "2CyiMVs0XRA5nFMdKvBdBZjgmUUqptlbs74OgvE2upms",
    },
  },
  Object {
    "dms_Location": Object {
      "owner": Object {
        "source": "GoogleDrive",
        "uid": "098765432109876543210",
      },
      "source": "GoogleDrive",
      "type": Object {
        "category": "Personal",
        "detail": "My Drive",
      },
      "uid": "people/098765432109876543210",
    },
  },
  Object {
    "dms_Location": Object {
      "owner": Object {
        "source": "GoogleDrive",
        "uid": "567890123456789012345",
      },
      "source": "GoogleDrive",
      "type": Object {
        "category": "Personal",
        "detail": "My Drive",
      },
      "uid": "people/567890123456789012345",
    },
  },
]
`;
